diff -ur a/lib/dbm_prog.c b/lib/dbm_prog.c
--- a/lib/dbm_prog.c	2026-05-20 15:52:10.047204118 +0700
+++ b/lib/dbm_prog.c	2026-05-20 15:53:37.937234439 +0700
@@ -3,11 +3,12 @@
 #include <sys/types.h>
 #include <limits.h>
 #include <string.h>
+#include<unistd.h>
 #include <fcntl.h>
 #ifdef FBSD_DATABASE
 #include <db.h>
 #else
-#include <ndbm.h>
+#include <gdbm-ndbm.h>
 #endif
 
 #ifdef FBSD_DATABASE	/* this needs to be explicit  */
diff -ur a/lib/Makefile b/lib/Makefile
--- a/lib/Makefile	2026-05-20 15:52:10.047204118 +0700
+++ b/lib/Makefile	2026-05-20 15:52:50.232125391 +0700
@@ -25,7 +25,7 @@
 # Linux users
 CC = gcc
 CFLAGS = -g -Wall -pedantic
-LIBS = -lm -lgdbm
+LIBS = -lm -lgdbm -lgdbm_compat
 ############################################################################
 
 OBJ = dbm_prog.c
@@ -34,12 +34,12 @@
 all: lexicon 
 
 lexicon: text710.edin dbm_prog
-	/bin/rm -i lexicon*
+	/bin/rm -f lexicon*
 	./dbm_prog -o lexicon -i text710.edin
-	touch lexicon
+#	touch lexicon
 
 text710.edin: text710.dat read_ox_dict
-	read_ox_dict text710.dat > text710.edin
+	./read_ox_dict text710.dat > text710.edin
 
 dbm_prog: $(OBJ)
 	$(CC) $(CFLAGS) -o $@ $(OBJ) $(LIBS)
diff -ur a/lib/read_ox_dict b/lib/read_ox_dict
--- a/lib/read_ox_dict	2026-05-20 15:52:10.047204118 +0700
+++ b/lib/read_ox_dict	2026-05-20 15:52:50.232125391 +0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 # Copyright (c) 1995 Alistair Conkie. All rights reserved.
 # This program is free software; you can redistribute it and/or
diff -ur a/src/Makefile b/src/Makefile
--- a/src/Makefile	2026-05-20 15:52:10.051204209 +0700
+++ b/src/Makefile	2026-05-20 15:52:50.232125391 +0700
@@ -25,7 +25,7 @@
 # Linux users
 CC = gcc
 CFLAGS = -g -Wall -pedantic
-LIBS = -lm -lgdbm
+LIBS = -lm -lgdbm -lgdbm_compat
 ############################################################################
 
 
diff -ur a/src/regerror.c b/src/regerror.c
--- a/src/regerror.c	2026-05-20 15:52:10.051204209 +0700
+++ b/src/regerror.c	2026-05-20 16:24:14.849828470 +0700
@@ -2,8 +2,7 @@
 #include "t2s.h"
 
 void
-regerror(s)
-     const char *s;
+regerror(const char *s)
 {
 #ifdef ERRAVAIL
   error("regexp: %s", s);
diff -ur a/src/regexp.c b/src/regexp.c
--- a/src/regexp.c	2026-05-20 15:52:10.051204209 +0700
+++ b/src/regexp.c	2026-05-20 16:22:13.102435270 +0700
@@ -195,8 +195,7 @@
  * of the structure of the compiled regexp.
  */
 regexp *
-regcomp(exp)
-     const char *exp;
+regcomp(const char *exp)
 {
   register regexp *r;
   register char *scan;
@@ -781,12 +780,10 @@
  - regexec - match a regexp against a string
  */
 int
-regexec(prog, string)
-     register const regexp *prog;
-     register const char *string;
+regexec(const regexp *prog, const char *string)
 {
   register char *s;
-  extern char *strchr();
+  //  extern char *strchr();
 
   /* Be paranoid... */
   if (prog == NULL || string == NULL) {
@@ -885,7 +882,7 @@
 {
   register char *scan;	/* Current node. */
   char *next;		/* Next node. */
-  extern char *strchr();
+  //  extern char *strchr();
 
   scan = prog;
 #ifdef DEBUG
diff -ur a/src/regexp.h b/src/regexp.h
--- a/src/regexp.h	2026-05-20 15:52:10.051204209 +0700
+++ b/src/regexp.h	2026-05-20 16:24:25.610127335 +0700
@@ -15,7 +15,7 @@
 	char program[1];	/* Unwarranted chumminess with compiler. */
 } regexp;
 
-extern regexp *regcomp();
-extern int regexec();
-extern void regsub();
-extern void regerror();
+extern regexp *regcomp(const char *exp);
+extern int regexec(const regexp *prog, const char *string);
+extern void regsub(const regexp *prog, const char *source, char *dest);
+extern void regerror(const char *s);
diff -ur a/src/regsub.c b/src/regsub.c
--- a/src/regsub.c	2026-05-20 15:52:10.051204209 +0700
+++ b/src/regsub.c	2026-05-20 16:23:20.668321106 +0700
@@ -30,17 +30,14 @@
  - regsub - perform substitutions after a regexp match
  */
 void
-regsub(prog, source, dest)
-     const regexp *prog;
-     const char *source;
-     char *dest;
+regsub(const regexp *prog, const char *source, char *dest)
 {
   register char *src;
   register char *dst;
   register char c;
   register int no;
   register int len;
-  extern char *strncpy();
+  //  extern char *strncpy();
 
   if (prog == NULL || source == NULL || dest == NULL) {
     regerror("NULL parm to regsub");
diff -ur a/src/space.c b/src/space.c
--- a/src/space.c	2026-05-20 15:52:10.051204209 +0700
+++ b/src/space.c	2026-05-20 15:52:50.232125391 +0700
@@ -15,7 +15,7 @@
 #ifdef FBSD_DATABASE
 #include <db.h>
 #else
-#include <ndbm.h>
+#include <gdbm-ndbm.h>
 #endif
 #include <fcntl.h>
 
diff -ur a/src/transcribe.c b/src/transcribe.c
--- a/src/transcribe.c	2026-05-20 15:52:10.051204209 +0700
+++ b/src/transcribe.c	2026-05-20 15:52:50.232125391 +0700
@@ -13,7 +13,7 @@
 #ifdef FBSD_DATABASE
 #include <db.h>
 #else
-#include <ndbm.h>
+#include <gdbm-ndbm.h>
 #endif
 #include <fcntl.h>
 
